### PR TITLE
Remove the option to include / exclude logged in users when the Analytics snippet is not placed via Site Kit (2910).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #2910

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by tofumatt** [comment]
@googlebot I consent.

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -119,6 +119,11 @@ public function register() {
 	 * @return bool
 	 */
 	protected function is_tracking_disabled() {
+		$settings = $this->get_settings()->get();
+		if ( ! apply_filters( 'googlesitekit_allow_tracking_disabled', $settings['useSnippet'] ) ) {
```
Comment (tofumatt) [inline includes/Modules/Analytics.php:123]: It's good to mention that a filter is documented elsewhere, so something like this should do the trick:

```suggestion
		// This filter is documented below, later in this function.
		if ( ! apply_filters( 'googlesitekit_allow_tracking_disabled', $settings['useSnippet'] ) ) {
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
This looks good, but if I'm understanding the issue the aim is that _a site with an existing tag placed by something other than Site Kit does not show the "Exclude from Analytics" option_. Correct?

If so, I think the code for this story should be updated: https://google.github.io/site-kit-wp/storybook/pull/3494/?path=/story/analytics-module-settings--edit-with-existing-tag-w-access

Right now it shows the existing tag option when the user has already placed a tag, which seems wrong 🙂

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 2]
```diff
@@ -41,6 +41,7 @@ import {
 } from '../common';
 import GA4PropertySelect from '../../../analytics-4/components/common/PropertySelect';
 import StoreErrorNotices from '../../../../components/StoreErrorNotices';
+import GA4Notice from '../common/GA4Notice';
```
Comment (tofumatt) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:44]:
```suggestion
```

File: includes/Modules/Analytics.php (hunk 1 above)
Comment (johnPhillips) [reply includes/Modules/Analytics.php:123]: You mean direct the reader to `Tag_Manager:: filter_analytics_allow_tracking_disabled`?

Comment (tofumatt) [reply includes/Modules/Analytics.php:123]: Yup! 🙂  Sorry, I thought it was in this file, but yes, pointing to there is good.

**Review by tofumatt — COMMENTED** [review]
I just had a call with @johnPhillips and it turns out I wasn't 100% clear on the issue from reading #2910's summary/ACs/IB. Turns out the implementation here is right and the Storybook stories are still correct/fixed, but I got confused about the conditions they were looking for.

After the merge conflict is resolved I'll take another look but I think my concerns about Storybook stories from the previous review were incorrect!

**Review by tofumatt — APPROVED** [review]

**Pull request merged**